uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -135,6 +135,7 @@ def send_500_response(self):
             msg,
         ]
         self.transport.write(b"".join(content))
+        self.handshake_started_event.set()
```

Comment:
Should we add a comment about why we're doing this? :)

For example...?

"Allow handler task to terminate cleanly, as websockets doesn't cancel it by itself."

Reply:
yes it's the kind of stuff you easily forget why it was put there ! I also added  a reference to te issue.